yarn-project/circuit-types/src/logs/function_l2_logs.ts

```diff
@@ -85,10 +87,13 @@ export class EncryptedFunctionL2Logs extends FunctionL2Logs<EncryptedL2Log> {
   /**
    * Creates a new L2Logs object with `numLogs` logs.
    * @param numLogs - The number of logs to create.
-   * @param logType - The type of logs to generate.
+   * @param test - Whether this call is a test, and won't overflow the kernels.
    * @returns A new EncryptedFunctionL2Logs object.
    */
-  public static random(numLogs: number): EncryptedFunctionL2Logs {
+  public static random(numLogs: number, test = false): EncryptedFunctionL2Logs {
```

Comment:
I think I will remove the flag - it's more about forcing it to fail here rather than failing deep into a simulation when we overflow the logs limit, so it's easier to debug a test. Some serialise/deserialise tests purposefully add tons of logs to ensure the reconstruction works properly, but I agree the flag is weird, so I'll reduce the num logs in those tests.